src/standard/image.rs

```diff
@@ -669,6 +704,85 @@ impl<E: OclPrm> Image<E> {
         Ok(new_img)
     }
 
+    /// Returns a new `Image` from an existant GL texture2D/3D.
+    // [WORK IN PROGRESS]
+    pub fn from_gl_texture(queue: &Queue, flags: MemFlags, image_desc: ImageDescriptor)
+            -> OclResult<Image<E>>
+    {
+        // FIXME need these defines !
+        // https://www.khronos.org/registry/cl/specs/opencl-1.x-latest.pdf#page=280
+        let texture_target = 0;
+        let miplevel = 0;
+        let texture = 0;
+
+        let obj_core = match image_desc.image_depth {
+            2 => unsafe { try!(core::create_from_gl_texture_2d(
+                                queue.context_core_as_ref(),
+                                texture_target,
+                                miplevel,
+                                texture,
+                                flags)) },
+            3 => unsafe { try!(core::create_from_gl_texture_3d(
+                                queue.context_core_as_ref(),
+                                texture_target,
+                                miplevel,
+                                texture,
+                                flags)) },
+            _ => unimplemented!() // FIXME: return an error ? or panic! ?
+        };
+
+        // FIXME can I do this from a GLTexture ?
+        let pixel_element_len = match core::get_image_info(&obj_core, ImageInfo::ElementSize) {
+            ImageInfoResult::ElementSize(s) => s / mem::size_of::<E>(),
+            ImageInfoResult::Error(err) => return Err(*err),
+            _ => return OclError::err("ocl::Image::element_len(): \
+                Unexpected 'ImageInfoResult' variant."),
+        };
+
+        let dims = [image_desc.image_width, image_desc.image_height, image_desc.image_depth].into();
+
+        let new_img = Image {
+            obj_core: obj_core,
+            queue: queue.clone(),
+            dims: dims,
+            pixel_element_len: pixel_element_len,
+            _pixel: PhantomData,
+        };
+
+        Ok(new_img)
+    }
+
+    /// Returns a new `Image` from an existant renderbuffer.
+    // [WORK IN PROGRESS]
+    pub fn from_gl_renderbuffer(queue: &Queue, flags: MemFlags, image_desc: ImageDescriptor,
```

Comment:
Let me be more clear... I'm not sure what you mean by which parameters... Parameters for what?